source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -191,11 +193,11 @@ async Task<DataStream> DownloadFileAction(CancellationToken ct)
                     clientOptions,
                     logger);
 
-                var orchestrator = await factory.CreateOrchestrator(scriptExecutionCancellationToken);
+                var orchestrator = factory.CreateOrchestrator(scriptServiceVersion);
```

Comment:
@sburmanoctopus Could we make it so that the "in memory" version of execute a scripts calls the below methods. That way the client itself gives an example of how to call the methods below and we always test those methods.

Reply:
I don't think I follow this one sorry.

The below methods are the publicly facing versions, which will have their own issues with metrics gathering etc. So I would not be inclined to call them from other methods.

They do call the same executors, so end up calling the same code.

It would be nice to have something show you how they should be called. But that's all done within the orchestrator.

Maybe this one is worth a chat?